qabelbox/src/test/java/de/qabel/qabelbox/identity/IdentityUseCaseTest.kt

```diff
@@ -0,0 +1,61 @@
+package de.qabel.qabelbox.identity
+
+import com.nhaarman.mockito_kotlin.mock
+import com.nhaarman.mockito_kotlin.verify
+import de.qabel.core.extensions.CoreTestCase
+import de.qabel.core.extensions.createIdentity
+import de.qabel.core.repository.inmemory.InMemoryIdentityRepository
+import de.qabel.qabelbox.BuildConfig
+import de.qabel.qabelbox.QblBroadcastConstants.Identities.*
+import de.qabel.qabelbox.SimpleApplication
+import de.qabel.qabelbox.eq
+import de.qabel.qabelbox.identity.interactor.MainIdentityUseCase
+import de.qabel.qabelbox.listeners.ActionIntentSender
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.robolectric.RobolectricGradleTestRunner
+import org.robolectric.annotation.Config
+
+@RunWith(RobolectricGradleTestRunner::class)
+@Config(application = SimpleApplication::class, constants = BuildConfig::class)
+class IdentityUseCaseTest() : CoreTestCase {
+
+    val identity = createIdentity("Alice")
+    val identityRepo = InMemoryIdentityRepository().apply { save(identity) }
+
+    val actionSender: ActionIntentSender = mock()
+    val useCase = MainIdentityUseCase(identityRepo, actionSender)
+
+    @Test
+    fun testLoadIdentity() {
+        val loaded = useCase.getIdentity(identity.keyIdentifier).toBlocking().value()
+        loaded eq identity
+    }
+
+    @Test
+    fun testLoadIdentities() {
+        val second = createIdentity("Bob").apply { identityRepo.save(this) }
+        val identities = useCase.getIdentities().toBlocking().value()
+        identities.identities.size eq 2
+        identities.identities eq setOf(identity, second)
+    }
+
+    @Test
+    fun testUpdateIdentity() {
+        identity.alias = "Forentroll"
+        identity.email = "forentroll@banane.de"
+        identity.phone = "12345678910"
+        useCase.saveIdentity(identity).toBlocking().value()
+        verify(actionSender).sendActionIntentBroadCast(IDENTITY_CHANGED, Pair(KEY_IDENTITY, identity))
+        val updated = identityRepo.find(identity.keyIdentifier)
+        updated eq identity
```

Comment:
Looks like this doesn't test anything. If the InMemoryIdentityRepository keeps instances of the identities, then this eq check would even work if useCase.saveIdentity(identity) would be empty because `identity` and `updated` are not only equal but are the same instance. This should work, too:
```KOTLIN
identity.alias = "some alias"
val updated = identityRepo.find(identity.keyIdentifier)
updated eq identity

identity.alias = "new alias"
updated eq identity
```